tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/FmiIntegrationTests.cs

```diff
@@ -0,0 +1,441 @@
+                                                    .ConfigureAwait(false);
+
+            //Assert
+            AssertResults(authResult,
+                          confidentialApp,
+                          "expectedInternalCacheKey",
+                          "expectedAudience");
+        }
+
+        [TestMethod]
+        [Ignore("Waiting for ESTS Implementation")]
+        //FMI-FIC -> Access token (2nd leg of FMI connector flow)
+        public async Task Flow7_FmiToken_From_FmiFicTestAsync()
+        {
+            //Arrange
+            X509Certificate2 cert = CertificateHelper.FindCertificateByName(TestConstants.AutomationTestCertName);
+            string expectedExternalCacheKey = null;
+
+            Action<TokenCacheNotificationArgs> extCacheKeyEvaluator = (args) =>
+            {
+                if (expectedExternalCacheKey != null)
+                {
+                    Assert.IsTrue(args.SuggestedCacheKey.Contains(expectedExternalCacheKey));
+                }
+            };
+
+            //Fmi app/scenario parameters
+            var clientId = "AppId to impersonate";
+            var scope = "api://d796a5d2-0fb6-499a-b311-8bf5b3d058f7/.default";
+
+            //Act
+            //Create application
+            var confidentialApp = ConfidentialClientApplicationBuilder
+                        .Create(clientId)
+                        .WithAuthority("https://login.microsoftonline.com/f645ad92-e38d-4d1a-b510-d1b09a74a8ca", true)
+                        .WithExtraQueryParameters("dc=ESTS-PUB-SCUS-LZ1-FD000-TEST1") //Enables MSAL to target ESTS Test slice
+                        .WithExperimentalFeatures(true) //WithFmiPath is experimental so experimental features needs to be enabled on the app
+                        .WithClientAssertion((options) => GetParentCredential(options))
+                        .BuildConcrete();
+
+            //Configure token cache serialization
+            confidentialApp.AppTokenCache.SetBeforeAccess(BeforeCacheAccess);
+            confidentialApp.AppTokenCache.SetAfterAccess(AfterCacheAccess);
+
+            //Recording test data for Asserts
+            var appCacheAccess = confidentialApp.AppTokenCache.RecordAccess(extCacheKeyEvaluator);
+            expectedExternalCacheKey = "";
+
+            //Acquire Fmi Cred
+            var authResult = await confidentialApp.AcquireTokenForClient(new[] { scope })
+                                                    .WithFmiPath("") //Sets fmi path in client credential request.
+                                                    .ExecuteAsync()
+                                                    .ConfigureAwait(false);
+
+            //Assert
+            AssertResults(authResult,
+                          confidentialApp,
+                          "expectedInternalCacheKey",
+                          "expectedAudience");
+        }
+
+        private static async Task<string> GetParentCredential(AssertionRequestOptions options)
```

Comment:
Added the assertion logic to a private method. It seemed like too much code to have this in every test. This is also how JM has it in his spec